src/highlight/highlightUtil.ts

```diff
@@ -1,5 +1,12 @@
-import { Annotation, AnnotationHighlight, Rect, Type } from '../@types';
+import { Annotation, AnnotationHighlight, DOMRectMini, Rect, Type } from '../@types';
 import { checkValue } from '../utils/util';
+import { getPageNumber } from '../document/docUtil';
+
+type SelectionItem = {
```

Comment:
It looks like this is duplicated from the store. Can the utilities make all the DOM API calls and then pass the resulting DOMRectMini and location values to an action that construct this state shape?

Reply:
Currently, it is utility that makes all the DOM API calls. And the return object of `getSelectionItem` is all the action needs. I can move that type inline with the function signature to avoid this duplicate type declaration.

Reply:
The type would still be duplicated, just inline. The code structure needs to be rethought and probably broken up to fully separate the DOM access and the state construction.